src/__tests__/App.test.tsx

```diff
@@ -93,20 +96,46 @@ describe('displaying cards', () => {
     await userEvent.click(saveButton)
     expect(mockPostCardsService).toHaveBeenCalled()
     await waitFor(() => {
-      expect(getByText(newCard.front)).toBeInTheDocument
-      expect(getByText(newCard.back)).toBeInTheDocument
+      expect(getByText(newCard.front)).toBeInTheDocument()
+      expect(getByText(newCard.back)).toBeInTheDocument()
+    })
+  })
+
+  it('It should delete flashcard from the list when clicking on Trash icon', async () => {
+    mockDeleteCardsService.mockResolvedValue(startCards[0])
+    const { getByText, getByTestId, queryByText } = render(<App />)
+
+    await waitFor(() => {
+      expect(getByText(startCards[0].front)).toBeInTheDocument()
+    })
+    const cardHeader = getByText(startCards[0].front)
+    const cardElement = cardHeader.parentElement as HTMLElement
+    expect(cardElement).toBeInTheDocument()
+    const editButton = cardElement?.querySelector('button') as HTMLElement
+    expect(editButton).toBeInTheDocument()
+
+    userEvent.click(editButton)
+    await waitFor(() => {
+      expect(getByTestId('delete')).toBeInTheDocument()
+    })
+    const deleteButton = getByTestId('delete')
+    userEvent.click(deleteButton)
+
+    await waitFor(() => {
+      expect(queryByText(startCards[0].back)).not.toBeInTheDocument()
```

Comment:
**nitpick(non-blocking):** I would check if amount of cards is different that initial amount. Eg:

```ts
  expect(cardElements.length)).toHaveLength(startCards.length - 1)
```